In [1]:
pip install beautifulsoup4

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [2]:
pip install html5lib

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [8]:
pip install requests

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [10]:
import requests
import time
from bs4 import BeautifulSoup

start = 0
results = []
#SEARCH_QUERIES = [None, 'financial spreadsheet', 'budget template', 'invoice template', 'finance', 'budget', 'business', 'invoice', 'curriculum', 'template', 'payslip', 'tax', 'receipt', 'contract', 'proposal', 'agreement', 'report', 'statement', 'analysis', 'data', 'database', 'payroll', 'resume', 'cv', 'application', 'form', 'survey', 'questionnaire', 'checklist', 'policy', 'procedure', 'guide', 'manual', 'presentation', 'briefing', 'memo', 'email'] 952

#SEARCH_QUERIES = [None, 'financial spreadsheet', 'budget template', 'invoice template', 'finance', 'budget', 'business', 'invoice', 'curriculum', 'template', 'payslip', 'tax']
#SEARCH_QUERIES = ['финансовая таблица', 'шаблон бюджета', 'шаблон счета-фактуры', 'финансы', 'бюджет', 'бизнес', 'счет-фактура', 'учебный план', 'шаблон', 'заработная плата', 'налог']
#SEARCH_QUERIES = ['جدول بيانات مالية', 'قالب ميزانية', 'قالب فاتورة', 'تمويل', 'ميزانية', 'أعمال', 'فاتورة', 'منهاج دراسي', 'قالب', 'كشف راتب', 'ضريبة']
#SEARCH_QUERIES = ['财务电子表格', '预算模板', '发票模板', '财务', '预算', '商业', '发票', '课程表', '模板', '工资单', '税']
#SEARCH_QUERIES = ['tax', 'receipt', 'contract', 'proposal', 'agreement', 'report', 'statement', 'analysis', 'data', 'database']
#SEARCH_QUERIES = ['database', 'payroll', 'resume', 'cv', 'application', 'form', 'survey', 'questionnaire', 'checklist', 'policy', 'procedure', 'guide', 'manual', 'presentation', 'briefing', 'memo', 'email']
#SEARCH_QUERIES = ['policy', 'procedure', 'guide', 'manual', 'presentation', 'briefing', 'memo', 'email']

#SEARCH_QUERIES = ['جدول بيانات مالية', 'قالب ميزانية', 'قالب فاتورة', 'تمويل', 'ميزانية', 'أعمال', 'فاتورة', 'منهاج دراسي', 'قالب', 'كشف راتب', 'ضريبة', 'إيصال', 'عقد', 'مقترح', 'اتفاقية', 'تقرير', 'بيان', 'تحليل', 'بيانات', 'قاعدة بيانات', 'كشف الرواتب', 'سيرة ذاتية', 'سيرة مهنية', 'تطبيق', 'نموذج', 'مسح', 'استبيان', 'قائمة تدقيق', 'سياسة', 'إجراء', 'دليل', 'كتيب', 'عرض تقديمي', 'إيجاز', 'مذكرة', 'بريد إلكتروني'] 3
#SEARCH_QUERIES = ['финансовая таблица', 'шаблон бюджета', 'шаблон счета-фактуры', 'финансы', 'бюджет', 'бизнес', 'счет-фактура', 'учебный план', 'шаблон', 'заработная плата', 'налог', 'квитанция', 'контракт', 'предложение', 'соглашение', 'отчет', 'заявление', 'анализ', 'данные', 'база данных', 'зарплата', 'резюме', 'заявка', 'форма', 'опрос', 'анкета', 'чек-лист', 'политика', 'процедура', 'руководство', 'мануал', 'презентация', 'информационное сообщение', 'меморандум', 'электронное письмо'] 87
SEARCH_QUERIES = ['财务电子表格', '预算模板', '发票模板', '财务', '预算', '商业', '发票', '课程表', '模板', '工资单', '税', '收据', '合同', '建议书', '协议', '报告', '声明', '分析', '数据', '数据库', '工资单', '简历', '履历', '申请', '表格', '调查', '问卷', '清单', '政策', '程序', '指南', '手册', '演示文稿', '简报', '备忘录', '电子邮件'] #12



for query in SEARCH_QUERIES:
  resultsFinished = False
  start = 0
  previousLength = 0
  count = 0
  while not resultsFinished:
    url = f"https://www.google.com/search?hl=it&q={query}+intent%3Axlsm%20filetype%3Axlsm&start={start}"
    r = requests.get(url)
    soup = BeautifulSoup(r.content, 'html5lib')

    articles = soup.findAll('div', {'class' : 'egMi0 kCrYT'})
    
    for article in articles:
      anchor = article.find('a')
      link = str(anchor).split('=')[2]
      doc_link = link.split('&')[0]
      results.append(doc_link)
    
    start += 10
    time.sleep(1)
    print("Res", len(results), query)
    
    if (len(results) == previousLength):
      count += 1
    print(count)
    previousLength = len(results)
    
    if count > 2:
      resultsFinished = True
    
print(len(results))

Res 8 财务电子表格
0
Res 8 财务电子表格
1
Res 8 财务电子表格
2
Res 8 财务电子表格
3
Res 10 预算模板
0
Res 10 预算模板
1
Res 10 预算模板
2
Res 10 预算模板
3
Res 10 发票模板
0
Res 10 发票模板
1
Res 10 发票模板
2
Res 10 发票模板
3
Res 11 财务
0
Res 11 财务
1
Res 11 财务
2
Res 11 财务
3
Res 11 预算
0
Res 11 预算
1
Res 11 预算
2
Res 11 预算
3
Res 14 商业
0
Res 14 商业
1
Res 14 商业
2
Res 14 商业
3
Res 15 发票
0
Res 15 发票
1
Res 15 发票
2
Res 15 发票
3
Res 18 课程表
0
Res 18 课程表
1
Res 18 课程表
2
Res 18 课程表
3
Res 18 模板
0
Res 18 模板
1
Res 18 模板
2
Res 18 模板
3
Res 18 工资单
0
Res 18 工资单
1
Res 18 工资单
2
Res 18 工资单
3
Res 19 税
0
Res 19 税
1
Res 19 税
2
Res 19 税
3
Res 20 收据
0
Res 20 收据
1
Res 20 收据
2
Res 20 收据
3
Res 22 合同
0
Res 22 合同
1
Res 22 合同
2
Res 22 合同
3
Res 22 建议书
0
Res 22 建议书
1
Res 22 建议书
2
Res 22 建议书
3
Res 23 协议
0
Res 23 协议
1
Res 23 协议
2
Res 23 协议
3
Res 25 报告
0
Res 25 报告
1
Res 25 报告
2
Res 25 报告
3
Res 26 声明
0
Res 26 声明
1
Res 26 声明
2
Res 26 声明
3
Res 28 分析
0
Res 28 分析
1
Res 28 分析
2
Res 28 分析
3
Res 29 数据
0
Res 29 数据
1
Res 29 数据
2
Res 29 数据
3
Res 29 数据库
0
Res 29 数据库
1
Res 29 数据库
2
Res 29 数据库
3


In [11]:
all_results = []
results_noduplicate = list(dict.fromkeys(results))
all_results += results_noduplicate
print(len(all_results))
print(len(results_noduplicate))

129
129


In [12]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Recupera la versione attuale della lista di link

In [13]:
import pandas as pd

# Define the path to the folder in your Google Drive
folder_path = '/content/drive/MyDrive/Scraped_Links'

# Define the filename of the CSV file in the folder
filename = 'all_results.csv'

# Define the path to the CSV file in your Google Drive
csv_path = folder_path + '/' + filename

# Read the CSV file into a pandas DataFrame
df = pd.read_csv(csv_path)

# Convert the DataFrame to a list (and flatten it)

result_list = [item for sublist in df.values.tolist() for item in sublist]


Aggiorna la lista di link con la nuova search e tira via i doppioni

In [14]:
all_results = all_results + result_list

all_results

all_results = list(dict.fromkeys(all_results))
len(all_results)

1065

Libera il folder dalla vecchia versione della lista

In [15]:
import os

# Set the path to the folder in your Google Drive
folder_path = '/content/drive/MyDrive/Scraped_Links'

# Loop through all files in the folder and delete them
for filename in os.listdir(folder_path):
    file_path = os.path.join(folder_path, filename)
    if os.path.isfile(file_path):
        os.remove(file_path)


Carica i nuovi link

In [16]:
import json
import pandas as pd

# Write results to JSON file
with open("all_results.json", 'w') as f:
    # indent=2 is not needed but makes the file human-readable 
    # if the data is nested
    json.dump(all_results, f, indent=2) 

# Read results from JSON file and convert to DataFrame
with open("all_results.json", 'r') as f:
    all_results = json.load(f)
df = pd.DataFrame(all_results)

# Define the path to the folder in your Google Drive
folder_path = '/content/drive/MyDrive/Scraped_Links'

# Save the DataFrame as a CSV file in your Google Drive
csv_path = folder_path + '/all_results.csv'
df.to_csv(csv_path, index=False)

print('CSV file saved to:', csv_path)


CSV file saved to: /content/drive/MyDrive/Scraped_Links/all_results.csv


La query completa tradotta (con google traduttore) in [Arabo,Russo,Cinese] ha portato rispettivamente [3,87,12] risultati nuovi. Per un totale di 1065 file excel.